view/src/main/webapp/WEB-INF/jsp/contestmanagement/manager/overviewTab.jspx

```diff
@@ -265,12 +265,14 @@
 			selectAllCheckboxElement.addEventListener("change", function (ev) {
 				var selectAllChecked = selectAllCheckboxElement.checked;
 				var contestsTableBody = document.getElementById('contestOverviewBody');
-				[].forEach.call(contestsTableBody.getElementsByClassName("checkbox"), function (element) {
-					var parentRow = getClosest(element, "tr");
-					if( parentRow.getAttribute("data-filter-visible")  === 'true'){
-						element.checked = !!selectAllChecked;
-					}
-				});
+                [].forEach.call(contestsTableBody.getElementsByTagName('input'), function (element) {
+                    if(element.getAttribute('type')=='checkbox' && element.getAttribute('name').includes("selectedContests")){
```

Comment:
In checking the name attribute, you depend on the name (which is primarily used for form binding  never changing. I would add a new extra class the checkboxes and then use `getElementsByClassName` to select those checkboxes. We usually prefix these classes with `js-` to clearly mark them as classes used for JavaScript. The class could be something like `js-contestSelectionCheckbox`.

Reply:
Good point, I will get to that.